### DISCUSSION TITLE
block specific users country

### DISCUSSION THREAD

**Discussion by AmoAlireza** [XTLS/Xray-core#3394] [Q&A] [answered]
hello

any possible way in **_config file_(json config)** to block users from specific country like iran ip range to connect the vpn?
like if user from iran and wants to connect vpn the xray core detect the user ip and and block it.

or reverse, like if user from specific country iran only can access to vpn and connect.

**Comment by kingwilliam** [comment]
There are many ways to accomplish your goals. Below is an example.

https://www.ip2location.com/free/visitor-blocker

how to use it? depends on you

![iranip](https://github.com/XTLS/Xray-core/assets/3607762/a1d1cc17-bc82-4cbd-a6d1-aed727c47ec4)

**Reply by AmoAlireza** [reply]
as i said, should be on json file, i know these ways.

**Comment by kingwilliam** [comment]
same answer, There are many ways to accomplish your goals. 

1. download iran ip and output cidr format and put it all

```
routing:
   {
      source: {
          *iran ip cidr format*
      }
      outbound: blocked
   }
```

2. use [Loyalsoldier](https://github.com/Loyalsoldier/geoip)/geoip generate yourself_iran.dat


```
routing:
   {
      source: {
          "ext:yourself_iran.dat:iran"
      }
      outbound: blocked
   }
```

**Reply by AmoAlireza** [reply, marked as answer]
thanks ❤
if anybody want to use, it's the complete example:

```
"routing": {
    "domainStrategy": "IPIfNonMatch",
    "rules": [
      {
        "type": "field",
        "source": [
          "YOUR_SOURCE_IP"
        ],
        "outboundTag": "blocked"
      }
    ]
  }
```
